pkg/corerp/frontend/controller/applications/graph_util.go

```diff
@@ -459,16 +464,10 @@ func outputResourcesFromAPIData(resource generated.GenericResource) []*corerpv20
 	return entries
 }
 
-// connectionsFromAPIData resolves the outbound connections for a resource from the generic resource representation.
-// For example if the resource is an 'Applications.Core/container' then this function can find it's connections
-// to other resources like databases. Conversely if the resource is a database then this function
-// will not find any connections (because they are inbound). Inbound connections are processed later.
-func connectionsFromAPIData(resource generated.GenericResource, allResources []generated.GenericResource) []*corerpv20231001preview.ApplicationGraphConnection {
+func resolveConnections(resource generated.GenericResource, jsonRefPath string, converter func(any) (string, corerpv20231001preview.Direction, error)) []*corerpv20231001preview.ApplicationGraphConnection {
```

Comment:
I refactored the code to use the same code for resolving connections, routes, and provides instead of repeating the same code.